File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function solution(numbers) {
    const getPermutations = (arr, selectNum) => {
        const result = [];
        if (selectNum === 1) return arr.map(el => [el]);
        
        arr.forEach((fixed, index, origin) => {
            const rest = [...origin.slice(0, index), ...origin.slice(index+1)];
            const permutations = getPermutations(rest, selectNum-1);
            const attached = permutations.map(el => [fixed, ...el]);
            result.push(...attached);
        });
        
        return result;
    }
    const isPrime = (num) => {
        if (num<2) return false;
        if (num===2) return true;
        
        for (let i=2; i<=Math.sqrt(num); i++) {
            if (num%i===0) return false;
        }
        return true;
    };
    
    const numbersArr = numbers.split('');
    const permutations = new Set();
    for (let i=1; i<=numbersArr.length; i++) {
        const result = getPermutations(numbersArr, i);
        result.forEach(v => {
            const number = Number(v.join(''));
            isPrime(number) && permutations.add(number);
        });
    };
    // console.log(permutations);
    return permutations.size;
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
import math
def solution(numbers):
    primes = set()
    visited = [False]*len(numbers)
    
    # numbers 리스트화
    arr = []
    for n in numbers:
        arr.append(n)
        
    def isPrime(num):
        if num<=1: return False
        if num==2: return True
        for i in range(2, int(math.sqrt(num))+1):
            if num%i==0:
                return False
        return True
    
    def DFS(num):
        if False not in visited:
            return
        # 소수 판별 후 DFS 재귀함수
        for i in range(len(arr)):
            if visited[i]:
                continue
            if isPrime(int(num+arr[i])):
                # print(num+arr[i])
                primes.add(int(num+arr[i]))
            visited[i]=True
            DFS(num+arr[i])
            visited[i]=False
    DFS("")
    
    return len(primes)